language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -235,6 +263,7 @@ impl DiemVM {
 
         charge_global_write_gas_usage(cost_strategy, &session, &txn_data.sender())?;
 
+        // TODO: question: why is `cost_strategy.disable_metering()` not called here?
```

Comment:
Totally besides this PR, but why `cost_strategy.disable_metering()` is not called in `execute_module` but is called in `execute_script_or_script_function`?

Reply:
Good catch. We can remove the `disable_metering` call -- the `success_transaction_cleanup` function creates a new `cost_strategy` that is not metered. And, that is the last remaining use of `disable_metering` so we can completely remove that function. I'll put up a separate PR for that change. It's not really part of this change.

Reply:
Thank you for the clarification Bob! I guess the `enable_metering` function can be removed too as I don't see a call site to that neither.

Reply:
Yes, I noticed that as well. I'm also noticing a few other related opportunities for simplifying this, so I'm going to review it.